src/lib/Libdb/db_postgres_attr.c

```diff
@@ -101,7 +101,13 @@ convert_array_to_db_attr_list(char *raw_array, pbs_db_attr_list_t *attr_list)
 	struct pg_array *array = (struct pg_array *) raw_array;
 	struct str_data *val = (struct str_data *)(raw_array + sizeof(struct pg_array));
 
-	if (ntohl(array->ndim) != 1 || ntohl(array->elemtype) != TEXTOID) {
+	if (ntohl(array->ndim) == 0) {
```

Comment:
As discussed raw_array contains an empty string in case  if either NULL or empty string itself is present in the attributes column of the corresponding table.
So we just need  to check this condition and decide accordingly.
So no need to convert raw_array to pg_array if this is the case and we can exit early from the function.

Reply:
I''m seeing empty in raw_array  even for NON NULL values. As we discussed we need to make use of APIs like PQgetisnull which might perfectly seem to apply for this case, but for which we need to study a little bit and decide. In the interest of time we will proceed with the current fix and since we are anyway planning to change db layer to handle null during which time we can take care  of these comments.